integration/service/create_test.go

```diff
@@ -309,6 +282,90 @@ func TestCreateServiceConfigFileMode(t *testing.T) {
 	assert.NilError(t, err)
 }
 
+func TestCreateServiceWithMemorySwap(t *testing.T) {
+	defer setupTest(t)()
+	d := swarm.NewSwarm(t, testEnv)
+	defer d.Stop(t)
+	client := d.NewClientT(t)
+	defer client.Close()
+
+	ctx := context.Background()
+	memoryLimit := int64(20 * units.MiB)
+	swap := memoryLimit + 1 * units.MiB
+
+	serviceID, task := createServiceWithSingleInstanceAndWaitConverge(t, d, client, ctx,
+		swarm.ServiceWithMemorySwap(swap, memoryLimit))
+
+	// verify that the container has the swap option set
+	ctnr, err := client.ContainerInspect(ctx, task.Status.ContainerStatus.ContainerID)
+	assert.NilError(t, err)
+	//assert.DeepEqual(t, ctnr.HostConfig.MemorySwap, swap)
```

Comment:
@anshulpundir that's the failing test. No timing issue at all :( it's set to -1 when failing...